app/PasswordPolicy.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace App;
+
+use ZxcvbnPhp\Zxcvbn;
+
+class PasswordPolicy
+{
+    private $zxcvbn;
+
+    public function __construct()
+    {
+        $this->zxcvbn = new Zxcvbn();
+    }
+
+    /**
+     * Validates the password based on Customer Portal settings.
+     * @param String $password Password to validate
+     * @return bool True if validation passed; otherwise, false.
+     */
+    public function isPasswordValid(String $password): bool
+    {
+        //The settings page was built with 1-5 instead of 0-4, fixing this here instead of making a migration.
+        $minimumPasswordStrength = SystemSetting::first()->password_strength_required - 1;
+        $zxcvbnResult = $this->zxcvbn->passwordStrength(substr($password, 0, 100));
```

Comment:
What is the purpose of limiting this to the 1st 100 characters?  Does not appear to be that the password fields are limited to that length that I could see.

Reply:
It's setup to match [the Sonar repo](https://github.com/SonarSoftwareInc/sonar/blob/develop/sonar/app/PasswordPolicy.php#L88). Removing the truncation and testing passwords with varying numbers of characters shows the performance degrades substantially with longer passwords.

16 averaged 0.003s
100 averaged 0.705s
400 averaged 5.117s